### PR TITLE
feat(webapp): UI for smart contract integration

### LINKED ISSUE #18: Show TX history for modified transaction
As a developer I want to show the history you have in order to make the necessary modifications to the smart contract

### LINKED ISSUE #19: Update previous transaction
As a developer I want to reclassify all the information inside the smart contracts. this in order to achieve a better use of the information and ensure that old data can be as up-to-date as possible and try to eliminate unclassified expenses as much as possible.

### LINKED ISSUE #21: UI Tool for adding tx metadata after they have been sent
As a developer I want to generate a UI-UX and a smart contract.
This tool can be used by the different delegates of the community to justify and classify the expenses

### LINKED ISSUE #23: EDAS Transfer Tool UI
As a developer I want to develop the UI for the smart contract and the information can be displayed in it.

### LINKED ISSUE #24: Interaction with UAL for EDAS Transfer Tool

### LINKED ISSUE #178: UI for smart contract integration
- [x]  Blockchain action 

- [x]  Uncategorized transaction table

- [x]  Form for categorize

### PR DESCRIPTION
**Pull request by warnerHurtado**
### What does this PR do?

- Resolve #178
- Resolve #18 
- Resolve #19 
- Resolve #24 
- Resolve #23
- Resolve #21 

### Steps to test

1. run the project and open localhost:3000 in your browser
2. click on spend tools and see the new features

#### CheckList

- [X] Follow proper Markdown format
- [X] The content is adequate
- [X] The content is available in both english and spanish
- [X] I Ran a spell check

### CODE REVIEW

File: webapp/src/hooks/customHooks/useForm.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const handleInputChange = ({ target }) => {
+    setValues({
+      ...values,
+      [target.name]: target.value
+    })
+
+    if (target.value.length > 0) {
+      const newErrors = { ...errors }
+      delete newErrors[target.name]
+      setErrors({ ...newErrors })
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useForm.js:19]: add a blank space before this line

File: webapp/src/hooks/customHooks/useForm.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+      setErrors({ ...errors, [target.name]: true })
+    }
+  }
+
+  const validateForm = form => {
+    let formErrors = {}
+    Object.keys(form).forEach(key => {
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useForm.js:27]: add a blank space before this line

File: webapp/src/hooks/customHooks/useForm.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      if (form[key].length < 1) {
+        formErrors = { ...formErrors, [key]: true }
+      } else {
+        const newErrors = { ...formErrors }
+        delete newErrors[key]
+        formErrors = { ...newErrors }
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useForm.js:33]: add a blank space before this line

File: webapp/src/hooks/customHooks/useForm.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      }
+    })
+    setErrors(formErrors)
+    return formErrors
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useForm.js:37]: add a blank space before this line

File: webapp/src/hooks/customHooks/useSpendToolsState.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
+import { useEffect, useState } from 'react'
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useSpendToolsState.js:1]: please reorder those imports by groups

File: webapp/src/hooks/customHooks/useSpendToolsState.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    errorsModal,
+    validateFormModal
+  ] = useForm({
+    newCategory: '',
+    newDescription: ''
+  })
+
+  const executeAction = async (data, account, name) => {
+    setErrorMessage('')
+    const transaction = {
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useSpendToolsState.js:44]: add a blank space before this line

File: webapp/src/hooks/customHooks/useSpendToolsState.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    setAmountCategorized(
+      categorizedTxData?.eden_transaction_aggregate?.aggregate?.sum?.amount || 0
+    )
+  }, [categorizedTxData])
+
+  useEffect(() => {
+    setAmountClaimed(
+      claimedTxData?.eden_transaction_aggregate?.aggregate?.sum?.amount || 0
+    )
+  }, [categorizedTxData])
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useSpendToolsState.js:138]: Why this useEffect has `categorizedTxData` as a dependency but never used?

the dependency should be `claimedTxData`

File: webapp/src/routes/SpendTools/index.js [hunk 8]
```diff
@@ -0,0 +1,358 @@
+import React, { memo } from 'react'
+import { makeStyles } from '@mui/styles'
+import Select from '@mui/material/Select'
+import Button from '@mui/material/Button'
+import { useTranslation } from 'react-i18next'
+import InputLabel from '@mui/material/InputLabel'
+import { IconButton, MenuItem, Modal, TextField, Tooltip } from '@mui/material'
```
Comment (tetogomez) [inline webapp/src/routes/SpendTools/index.js:7]: Try to use one level MUI import just for consistency 

in order to maintain the performance the idea is to use the top-level import

please check it: https://mui.com/material-ui/guides/minimizing-bundle-size/#option-one-use-path-imports

File: webapp/src/routes/SpendTools/styles.js [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+    padding: '8px 16px',
+    [theme.breakpoints.up('xs')]: {
+      maxWidth: '377px'
+    }
+  },
+  selectForm: {
+    alignSelf: 'stretch',
+    height: '48px',
+    borderRadius: '6px',
+    '& .css-1d3z3hw-MuiOutlinedInput-notchedOutline': {
```
Comment (tetogomez) [inline webapp/src/routes/SpendTools/styles.js:94]: I'm pretty sure if You use the correct MUI class you don't need to write that long css classes

example: 
`.MuiOutlinedInput-notchedOutline`

File: webapp/src/routes/SpendTools/styles.js [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+    },
+    '& #id-table-container': {
+      marginTop: '16px',
+      height: '368.5px'
+    }
+  },
+  buttonContainer: {
+    display: 'flex',
+    justifyContent: 'center',
+    '& .css-1exfphm-MuiButtonBase-root-MuiButton-root': {
```
Comment (tetogomez) [inline webapp/src/routes/SpendTools/styles.js:115]: same here

File: webapp/src/routes/SpendTools/styles.js [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+      width: '345px',
+      height: '48px',
+      backgroundColor: '#2563eb',
+      borderRadius: 0,
+      padding: '0 16px',
+      textTransform: 'none'
+    },
+    '& .css-1exfphm-MuiButtonBase-root-MuiButton-root:hover': {
```
Comment (tetogomez) [inline webapp/src/routes/SpendTools/styles.js:123]: and here

File: webapp/src/utils/format-with-thousand-separator.js [hunk 12]
```diff
@@ -8,6 +8,6 @@ export const formatWithThousandSeparator = (value, precision) => {
   if (precision >= 0) {
     newValue = newValue.toFixed(precision)
   }
-
-  return newValue.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+  const comma = newValue.toString().split('.')
```
Comment (tetogomez) [inline webapp/src/utils/format-with-thousand-separator.js:11]: REMEMBER blank space after const declaration

**Review by tetogomez — CHANGES_REQUESTED** [review]
Please check the comments

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**